llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -252,6 +252,12 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
   getActionDefinitionsBuilder({G_FCEIL, G_FFLOOR})
       .libcallFor({s32, s64});
 
+  getActionDefinitionsBuilder(G_JUMP_TABLE).legalFor({p0});
+
+  getActionDefinitionsBuilder(G_BRJT).legalIf([=](const LegalityQuery &Query) {
+    return typeIs(0, p0) && typeInSet(1, {s32, sXLen});
+  });
+
```

Comment:
```.
legalFor({{p0, s32}, {p0, sXLen}});
```

Reply:
As written this statment is always, you have to pass Query to typeIs and typeInSet. Other wise it's just checking if the address of the typeIs and typeInSet functions are non-null. Which of course they are.

Reply:
Does the s32 case ever occur? IRTranslator::emitJumpTableHeader seems to use the integer size of the pointer to create the type for that operand.